network/src/network.rs

```diff
@@ -282,26 +284,32 @@ impl NetworkState {
         self.local_private_key().to_peer_id().to_base58()
     }
 
-    pub(crate) fn listened_addresses(&self, count: usize) -> Vec<(Multiaddr, u8)> {
-        self.listened_addresses
+    pub(crate) fn listened_addrs(&self, count: usize) -> Vec<(Multiaddr, u8)> {
+        self.listened_addrs
             .read()
             .iter()
             .take(count)
             .map(|(addr, score)| (addr.to_owned(), *score))
             .collect()
     }
 
+    pub(crate) fn score_listened_addr(&self, addr: Multiaddr, inc_score: u8) {
```

Comment:
The function name `score_listened_addr` is confusing. I recommend `vote_listened_addr`, since votes are usually accumulated.

```suggestion
    pub(crate) fn vote_listened_addr(&self, addr: Multiaddr, votes: u8) {
```